src/runtime/entitlements-manager.js

```diff
@@ -366,17 +369,22 @@ export class EntitlementsManager {
     entitlementResult,
     entitlementSource,
     optionalToken = '',
-    optionalIsUserRegistered = null
+    optionalIsUserRegistered = null,
+    optionalSubscriptionTimestamp = null
   ) {
     const message = new EntitlementsRequest();
     message.setUsedEntitlement(usedEntitlement);
     message.setClientEventTime(toTimestamp(Date.now()));
     message.setEntitlementResult(entitlementResult);
     message.setEntitlementSource(entitlementSource);
     message.setToken(optionalToken);
+    message.setSubscriptionTimestamp(optionalSubscriptionTimestamp);
```

Comment:
Seems like this line (381) and lines 385-387 are doing the same thing a bit differently?

Reply:
Is the `if` statement necessary? Just curious, because I saw one on line 745 as well

Reply:
> Is the `if` statement necessary? Just curious, because I saw one on line 745 as well

I was trying not to break current path. I wasn't sure about an `undefined` or something else may break it.

Reply:
In that case, would it make sense to remove line 381?

Reply:
Sure, actually I'm not sure why I added line 381. It should be removed. Thanks!